secret/scanner.go

```diff
@@ -72,6 +141,34 @@ func (s Scanner) Scan(args ScanArgs) types.Secret {
 	}
 }
 
+func removeAllowedSecrets(allRuleResults []RuleResult, allowedLocations [][]int) []RuleResult {
+	if len(allowedLocations) > 0 {
+		results := make([]RuleResult, 0)
+		for _, result := range results {
+			found := false
+			if result.EndPosition < allowedLocations[0][0] {
+				results = append(results, result)
+				continue
+			}
+			for _, allowedLocation := range allowedLocations {
+				if allowedLocation[0] > result.EndPosition {
+					break
+				}
```

Comment:
I think it assumes `allowedLocations` is sorted, but it looks like it is not the case. There are two places to call `removeAllowedSecrets()` and iterate `AllowList.Regexes`. The result of merging results of `regex.FindAllIndex()` will not be sorted even though `regex.FindAllIndex()` returns a sorted slice.
https://github.com/aquasecurity/fanal/blob/6b9b47f53e3f9ef4af7e4c750fc37d9c8a726827/secret/scanner.go#L104-L106
https://github.com/aquasecurity/fanal/blob/6b9b47f53e3f9ef4af7e4c750fc37d9c8a726827/secret/scanner.go#L116-L118

Let's say the first `regex.FindAllIndex()` returns `[][]int{{10, 13}, {15, 17}}` and the second `regex.FindAllIndex()` returns `[][]int{{2, 4}, {6, 7}}`. Then, `allowedLocations` will be `[][]int{{10, 13}, {15, 17}, {2, 5}, {6, 7}}`. And `StartPosition` is 3 and `EndPosition` is 4, for example. `10 > 4` will be true and the result is added into `results`. But it is actually within `{2, 5}` and should be allowed.